packages/core/test/unit/application.test.ts

```diff
@@ -8,10 +8,60 @@ import {Application, Server, Component} from '../../index';
 import {Context, Constructor} from '@loopback/context';
 
 describe('Application', () => {
+  describe('controller binding', () => {
+    let app: Application;
+    class MyController {}
+
+    beforeEach(givenApp);
+
+    it('binds a controller', () => {
+      const binding = app.controller(MyController);
+      expect(binding.tags.has('controller')).to.be.true();
+      expect(binding.key).to.equal('controllers.MyController');
+      expect(app.findByTag('controller')[0]).to.equal(binding);
+    });
+
+    it('binds a controller with custom name', () => {
+      const binding = app.controller(MyController, 'my-controller');
+      expect(binding.tags.has('controller')).to.be.true();
+      expect(binding.key).to.equal('controllers.my-controller');
+      expect(app.findByTag('controller')[0]).to.equal(binding);
+    });
+
+    function givenApp() {
+      app = new Application();
+    }
+  });
+
+  describe('component binding', () => {
+    let app: Application;
+    class MyController {}
+    class MyComponent implements Component {
+      controllers = [MyController];
+    }
+
+    beforeEach(givenApp);
+
+    it('binds a component', () => {
+      app.component(MyComponent);
+      expect(app.findByTag('component').length).to.equal(1);
+    });
+
+    it('binds a component', () => {
+      app.component(MyComponent, 'my-component');
+      expect(app.find('components.my-component').length).to.equal(1);
+    });
+
+    function givenApp() {
+      app = new Application();
+    }
+  });
+
   describe('server binding', () => {
     it('defaults to constructor name', async () => {
       const app = new Application();
-      app.server(FakeServer);
+      const binding = app.server(FakeServer);
+      expect(binding.tags.has('server')).to.be.true();
```

Comment:
If `has` returns a false, then how are we going to find out what happened wrong? Here is a better check:

```ts
expect(Array.from(binding.tags)).to.containEql('server');
```

Also please use a single assertion per test and move the assertion checking binding tag to a new test.